### DISCUSSION TITLE
net metering

### DISCUSSION THREAD

**Discussion by tzapu** [emporia-vue-local/esphome#209] [Q&A] [answered]
i am confused by the net metering example, it also does not work properly in my case (solar + battery + 3 phase)

should net metering not add all 3 phases? so if i have export on phase 1 and 2, 2kW each and import 3kW on phase 3 the resulting total power is exporting 1kW ? 

thank you for this amazing project/effort!

**Comment by flaviut** [comment]
The net metering example splits a single clamp into two sensors:

- a power from grid sensor, where the filters contain `*pos`
- a power to grid sensor, where the filter contains `*invert`

I'm not sure if this all makes sense with the way home assistant works, I don't have net metering and this was a contribution.

If you'd like the behavior you describe, then I think removing any `*pos` or `*invert` filter would be the correct way to achieve it.

**Reply by tzapu** [reply]
thanks, advanced some more on my config, will post it on discussions when i m happy. 
right now i have a problem if the vue resets and restores and older value for daily totals, not sure how to best handle that as i think it might break statistics

**Reply by peterg79** [reply, marked as answer]
The net metering example simply measures import and export on each phases separately. This way it's in align with home-assistant's energy dashboard, where you can specify `Grid consumption` and `Return to grid` separately (see screenshot). For 3 phases you should be able to add a `phase_c` similarly. Having a battery does not make a difference here, as you measure everything on the incoming wires (battery solar charging does not show up as export, consuming from battery does not show up as import).

If you want to see net metering without home-assistant, then removing `*pos` and `*invert` should work.

<img width="363" alt="Screenshot 2023-09-22 at 11 45 02 PM" src="https://github.com/emporia-vue-local/esphome/assets/19354642/16fcc931-78d1-4ae2-97a3-495c62bbdbea">
<img width="408" alt="Screenshot 2023-09-22 at 11 49 12 PM" src="https://github.com/emporia-vue-local/esphome/assets/19354642/33829d1b-a06f-4846-99d2-8a993118dddc">
<img width="414" alt="Screenshot 2023-09-22 at 11 50 02 PM" src="https://github.com/emporia-vue-local/esphome/assets/19354642/53ce301d-9924-4bf6-8868-0b1bc4b946aa">

**Reply by tzapu** [reply]
ah, now i see, it delegates the net metering totals to home assistant basically, nice
did a config to have it computed in the vue in the end but that s an interesting solution as well, cheers

is there some way to see if i am overpowering the esp in the vue? calculating too much stuff, having it do more than it can?

**Reply by flaviut** [reply]
> is there some way to see if i am overpowering the esp in the vue? calculating too much stuff, having it do more than it can?

computers are very fast. unless you're repurposing the hardware to do something very different, this is simply not an issue

**Comment by flaviut** [comment]
Good discussion on that issue here:
https://github.com/emporia-vue-local/esphome/discussions/206
